binaries/download_binaries.py

```diff
@@ -0,0 +1,152 @@
+    """Downloads a file and write it to the file system."""
+
+    file_name = download_url.split('/')[-1]
+    # Don't try to download files other than the ones starting with
+    # one of the strings in `only_download_files_starting_with`.
+    if not file_name.startswith(tuple(only_download_files_starting_with)):
+      return
+    file_path = os.path.join(output_dir, file_name)
+    # Download with `stream=True` so not to use so much memory.
+    with requests.get(url=download_url, stream=True) as res:
+      # Was the connection successful?
+      try:
+        res.raise_for_status()
+      except Exception as ex:
+        raise RuntimeError('\'{}\' couldn\'t be downloaded\nException: {}\n'.format(
+            download_url, ex))
+      print('downloading', file_name)
+      with open(file_path, 'wb') as binary:
+        for content in res.iter_content(chunk_size=65536):
+          binary.write(content)
+      print(file_name, 'is downloaded.')
+    # Set executable permissions for the downloaded binaries.
+    default_permissions = 0o755
+    os.chmod(file_path, default_permissions)
+    # Add the this file we just downloaded to the downloaded files.
+    downloaded_files.append(file_name)
+
+  assets: List[dict] = []
+  for end_point in github_api_end_points:
+    res = requests.get(end_point,
+                       # This header is recommended by github docs.
+                       headers={'Accept': 'application/vnd.github.v3+json'})
+    res.raise_for_status()
+    res_in_json = res.json()
+    # `res_in_json[0]` will grab the latest release.
+    assets.extend(res_in_json[0]['assets'])
+
+  download_threads: List[Thread] = []
+  downloaded_files: List[str] = []
+
+  # Loop over all the assets and download them.
+  for asset in assets:
+    download_url: str = asset['browser_download_url']
+    download_threads.append(Thread(target=download_binary,
+                                   args=(download_url, downloaded_files)))
+    # Start downloading the asset.
+    download_threads[-1].start()
+
+  # Wait until all the binaries are downloaded.
+  while(any(download_thread.is_alive() for
+            download_thread in download_threads)):
+    time.sleep(1)
+
+  return downloaded_files
+
+
+def select_binaries(binaries_names: List[str],
+                    name_contains: List[str]) -> List[str]:
+  """A non-reliable method to automate selecting which binaries goes
+  for which platform."""
+
+  # The subset of the binaries we will select.
+  selected_binaries: List[str] = []
+  for binary_name in binaries_names:
+    # If we have every substring the `binary_name` so we can select it safely.
+    if all(substring.lower() in binary_name.lower() for
+           substring in name_contains):
+      selected_binaries.append(binary_name)
+  return selected_binaries
+
+
+def main():
+  # https://api.github.com/repos/{owner}/{repo}/releases
+  releases_urls = ['https://api.github.com/repos/joeyparrish/static-ffmpeg-binaries/releases',
```

Comment:
I think versioned download URLs could be constructed without resorting to the GitHub API.

Something like:

```py
PACKAGER_VERSION = 'v2.5.1'
FFMPEG_VERSION = 'early-trial-122`  # FIXME: update when first official release made

PACKAGER_DL_PREFIX = 'https://github.com/google/shaka-packager/releases/download/' + PACKAGER_VERSION
PACKAGER_DL = {
  'win-x64': PACKAGER_DL_PREFIX + '/packager-win.exe',
  'osx-x64': PACKAGER_DL_PREFIX + '/packager-osx',
}

# etc...
```

I think the download script could be much simpler this way.  Updates would just involve updating the tags at the top.